edx_arch_experiments/summaryhook_aside/block.py

```diff
@@ -0,0 +1,80 @@
+"""Xblock aside enabling OpenAI driven summariess"""
+
+from django.conf import settings
+from django.template import Context, Template
+from web_fragments.fragment import Fragment
+from xblock.core import XBlockAside
+
+from edx_arch_experiments.summaryhook_aside.waffle import summary_enabled
+
+summary_fragment = """
+<div class="summary-hook">
+  <div id="ai-spot-root"
+    data-url-api="{{data_url_api}}"
+    data-text-identifier="{{data_text_identifier}}"
+  ></div>
+  <script type="text/javascript" src="{{js_url}}" defer="defer"></script>
+</div>
+"""
+
+
+def _render_summary(context):
+    template = Template(summary_fragment)
+    return template.render(Context(context))
+
+
+def _children_have_summarizable_content(block):
+    """
+    Only if a unit contains HTML blocks with sufficient text in them
+    is it worth injecting the summarizer.
+    """
+    children = block.get_children()
+    for child in children:
```

Comment:
Note that this doesn't handle all your edge cases, because the children of vertical can be a nested structure, like [content experiments](https://edx.readthedocs.io/projects/open-edx-building-and-running-a-course/en/latest/course_features/content_experiments/content_experiments_add.html#define-the-experiment-content-in-the-split-test-file), where you have essentially nested verticals.

That being said, **I think what you're doing here is entirely appropriate for an experiment**, and I'm not advocating that you change it. You're covering the 99% use case, and doing it in a way that's simple and defensive. But it might be worth making a comment that there exist edge cases this will ignore.

Reply:
One other edge case you might run into with this is that very occasionally people copy-paste from MS Word and end up pasting in base64 encoded images as data URLs. That will result in this code path being triggered for HTML that has almost no real content (because the image just blows everything out). Again, not worth addressing in experimental code, but possibly worth noting for the future.

Reply:
Oh, actually that first thing means that you could potentially have an edge case where you have multiple copies of this injected into a Unit (one at the top Vertical level, one at the nested Vertical inside a content experiment). This is going to be *really uncommon*, since it would require there to be a lengthy text outside the experiment and text within the experiment. But it is possible, which means at some point, some content on edx.org is going to run across it. Again, not a problem for limited rollout I think. Just something to watch for.